File: Python/Observer/Blog.py
# -*- coding:utf-8 -*-

from Observer import *

class Blog:

    def __init__(self):
        self.__observerList = list()
        self._status = None

    def Attach(self, observer):
        self.__observerList.append(observer)

    def Detach(self, observer):
        self.__observerList.remove(observer)

    def Notify(self):
        for observer in self.__observerList:
            observer.Update()

    def SetStatus(self, str):
        self._status = str

    def GetStatus(self):
        return self._status

class BlogCSDN(Blog):

    def __init__(self, name):
        Blog.__init__(self)
        self.__blogName = name

    def SetStatus(self, str):
        self._status = self.__blogName + " " + str

File: CPlusPlus/Observer/src/Blog.cpp
/*************************************************************************
    > File Name: Blog.cpp
    > Author: ma6174
    > Mail: ma6174@163.com 
    > Created Time: 2017年03月22日 星期三 22时07分21秒
 ************************************************************************/

#include"Blog.h"
#include"ObserverBlog.h"

using namespace std;

Blog::~Blog()
{
	;
}

void Blog::Attach(shared_ptr<Observer> tempObserver)
{
	observerList.push_back(tempObserver);
}

void Blog::Detach(shared_ptr<Observer> tempObserver)
{
	observerList.remove(tempObserver);
}

void Blog::Notify()
{
	for(auto it=observerList.begin();it!=observerList.end();++it)
	{
		(*it)->Update();
	}
}

void Blog::SetStatus(const std::string& str)
{
	status=str;
}

string Blog::GetStatus()
{
	return status;
}
